plugins/wporg-5ftf/includes/email.php

```diff
@@ -37,6 +37,23 @@
 // Longer than `get_password_reset_key()` just to be safe. See https://core.trac.wordpress.org/ticket/43546#comment:34
 const TOKEN_LENGTH = 32;
 
+add_action( 'wp_head', __NAMESPACE__ . '\prevent_caching_auth_tokens', 99 );
+
+/**
+ * Prevent caching mechanisms from caching authentication tokens.
+ *
+ * Search engines would often be too slow to index tokens before they expire, but other mechanisms like Varnish,
+ * etc could create situations where they're leaked to others.
+ */
+function prevent_caching_auth_tokens() {
+	if ( ! isset( $_GET['auth_token'] ) && ! isset( $_POST['auth_token'] ) ) {
```

Comment:
Should this be an OR (`||`) scenario rather than `&&`? Rephrased, is `auth_token` going to be both in the GET and POST superglobals at the same time when it needs to prevent caching?
____
Also, you can combine `isset()` calls (but not `empty()` 🙃):

```
if ( ! isset( $_GET['auth_token'], $_POST['auth_token'] ) ) {
```

Reply:
The intent here is to only send `no-cache` when an auth token is provided somewhere in the request, regardless whether it's coming from `$_GET` or `$_POST`.

So, it seems like:

```php
if ( ! isset( $_GET['auth_token'] ) || ! isset( $_POST['auth_token'] ) ) {
    return;
}
```

...would return early when the token was provided, and `no-cache` would be sent when no token was provided. 

The `&&` worked in my tests, but I could be confusing something? What do you think?

Reply:
Ah, yeah you are right. I tripped up on the negatives. So this would be the same as:

```
if ( isset( $_GET['auth_token'] ) || isset( $_POST['auth_token'] ) ) {
    nocache_headers();
}
```

Ignore me 🙃

Reply:
That's a good point, though; maybe it'd be more readable in that form 🤔 

I typically like early returns rather than nesting, but in this case it probably hurts more than it helps. I'm gonna flip that to your version.